pkg/asset/installconfig/platformpermscheck.go

```diff
@@ -56,12 +56,17 @@ func (a *PlatformPermsCheck) Generate(dependencies asset.Parents) error {
 	case aws.Name:
 		permissionGroups := []awsconfig.PermissionGroup{awsconfig.PermissionCreateBase}
 		usingExistingVPC := len(ic.Config.AWS.Subnets) != 0
+		usingKMS := len(ic.Config.AWS.DefaultMachinePlatform.EC2RootVolume.KMSKeyARN) != 0
```

Comment:
1. I believe both `DefaultMachinePlatform` and `EC2RootVolume` could be `nil`, which is probably why the e2e tests are failing
2. You will probably need to check the other machine pools (master and worker)--not just the default machine pool.

Reply:
> I believe both DefaultMachinePlatform and EC2RootVolume could be nil, which is probably why the e2e tests are failing

Isn't this still a problem? Won't we have a nil pointer derefence if `DefaultMachinePlatform` or `EC2RootVolume` is nil? 

> You will probably need to check the other machine pools (master and worker)--not just the default machine pool.

Still need to check this. Actually, we just need to check the master machine pool. Worker permissions would be handled by operators.